tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-confusion-matrix.html

```diff
@@ -0,0 +1,256 @@
+              }
+            })
+            .attr('class', column => {
+              if (column == 0) {
+                return 'n-label';
+              } else {
+                return 'pred-label';
+              }
+            });
+
+        // Set the confusion matrix rows.
+        const rows = tbody.selectAll('tr')
+          .data(paddedCounts)
+          .enter()
+          .append('tr');
+        const cells = rows.selectAll('td')
+          .data((row, j) => {
+            return row.map((entry, i) => {
+              if (i == 0) {
+                return {value: 'Actual ' + labelsList[j]};
+              } else {
+                return {value: entry, row: j};
+              }
+            });
+          })
+          .enter()
+          .append('td').attr('class', (d, i) => {
+            if (i == 0) {
+              return 'actual-label';
+            } else if (d.value == 0) {
+              return '';
+            } else if (i - 1 == d.row) {
+              return 'correct';
+            } else if (i <= countMatrix.length) {
+              return 'incorrect';
+            } else {
+              return '';
+            }
+          }).style('background', (d, i) => {
+            if (i == 0 || i > countMatrix.length) {
+              return '#FFFFFF';
+            } else {
+              let c = d3.color('darkgrey');
+              c.opacity = d.value / total;
+              return c + '';
+            }
+          });
+        const cellDivs = cells.append('div').classed('flex', true);
+        cellDivs.append('div').classed('parenthetical', (d, i) => !!i)
+            .text(d =>
+              'row' in d ? '(' + d.value + ')' : '');
+        cellDivs.append('div').text(
+          (d, i) => i > 0 ? d3.format(",.1%")(d.value / total) : d.value);
+
+        // Set the footer row.
+        const footCells = tfoot.append('tr')
+          .selectAll('td')
+          .data(indices).enter()
+          .append('td');
+        footCells.classed('actual-label', (d, i) => i == 0);
+        const footCellDivs = footCells.append('div').classed('flex', true);
+        footCellDivs.append('div').classed('parenthetical', (d, i) => !!i).text(
+          column => {
+            if (column == 0) {
+              return 'Total';
+            } else if (column <= countMatrix.length) {
+              return '(' + columnTotals[column - 1] + ')';
+            } else {
+              return '';
+            }
+          });
+        footCellDivs.append('div').text(column => {
+          if (column == 0) {
+            return '';
+          } else if (column <= countMatrix.length) {
+            return d3.format(",.1%")(columnTotals[column - 1] / total);
+          } else {
+            return '';
+          }
+        });
+      },
+
+      /**
+       * Returns an list of indices from 0 to tableDim.
+       */
+      getInidices: function(tableDim) {
+        return Array.apply(null, {length: tableDim}).map(Number.call, Number);
```

Comment:
nit: nifty trick but can be hard to read. Would you say `[...new Array(tableDim)].map((_, i) => i)` read better?

Reply:
done